cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaReconcilerStatusTest.java

```diff
@@ -181,6 +184,87 @@ public void testKafkaReconcilerStatus(VertxTestContext context) {
         }));
     }
 
+    @Test
+    public void testKafkaReconcilerStatusUpdateVersion(VertxTestContext context) {
+        Kafka kafka = new KafkaBuilder(KAFKA)
+                .editOrNewSpec()
+                .editOrNewKafka()
+                .withReplicas(1)
+                .endKafka()
+                .endSpec()
+                .editOrNewStatus()
+                // supportedVersions is an ordered set, get the first element for the lowest kafka version
+                .withKafkaVersion(VERSIONS.supportedVersions().iterator().next())
+                .endStatus()
+                .build();
+
+        ResourceOperatorSupplier supplier = ResourceUtils.supplierWithMocks(false);
+
+        // Mock the secrets needed for Kafka client
+        SecretOperator mockSecretOps = supplier.secretOperations;
+        Secret secret = new Secret();
+        when(mockSecretOps.getAsync(eq(NAMESPACE), eq(KafkaResources.clusterCaCertificateSecretName(CLUSTER_NAME)))).thenReturn(Future.succeededFuture(secret));
+        when(mockSecretOps.getAsync(eq(NAMESPACE), eq(ClusterOperator.secretName(CLUSTER_NAME)))).thenReturn(Future.succeededFuture(secret));
+
+        // Run the test
+        KafkaReconciler reconciler = new MockKafkaReconcilerStatusTasks(
+                new Reconciliation("test-trigger", Kafka.RESOURCE_KIND, NAMESPACE, CLUSTER_NAME),
+                supplier,
+                kafka
+        );
+
+        KafkaStatus status = new KafkaStatus();
+
+        Checkpoint async = context.checkpoint();
+        reconciler.reconcile(status, Clock.systemUTC()).onComplete(context.succeeding(v -> context.verify(() -> {
+
+            // Check kafka version updated to default
+            assertThat(status.getKafkaVersion(), is(VERSIONS.defaultVersion().version()));
+
+            async.flag();
+        })));
+    }
+
+    @Test
+    public void testKafkaReconcilerStatusCustomKafkaVersion(VertxTestContext context) {
+        // supportedVersions is an ordered set, get the first element for the lowest kafka version
+        String lowestSupportedVersion = VERSIONS.supportedVersions().iterator().next();
+        Kafka kafka = new KafkaBuilder(KAFKA)
+                .editOrNewSpec()
+                .editOrNewKafka()
+                .withVersion(lowestSupportedVersion)
+                .withReplicas(3)
+                .endKafka()
+                .endSpec()
+                .build();
```

Comment:
Same comments as above ... would it be better to use the constant and indent the builder?